### DISCUSSION TITLE
Issues with Bitwarden Directory Connector and AD groups not applying to users after a second sync.

### DISCUSSION THREAD

**Discussion by Sir-IT-Papi** [dani-garcia/vaultwarden#4514] [Q&A]
Hello! I'm not sure if this is a bug or if I'm doing something wrong, but I'm currently setting up VW for my org, but I just came across this issue, so I initially ran a sync using BWDC with a select few users and these users are all a part of the same group on AD, so when I set up the users I finally managed to sync their AD group and when I checked on VW it automatically applied that group to those users. Later I added more users to test it out from the same group and I noticed that these users weren't automatically applied to the group like the previous user, so I thought "let me resync the group maybe it needs to be applied again after the user is already in" after that it removed all the users from that group and since then I haven't been able to get users to automatically be assigned to their appropriate AD groups. We have like 20 AD groups and 300+ users and ideally it would be nice if the users are automatically applied to their appropriate groups. 

Thank you!

**Comment by jobritz** [comment]
I think I had the same issue, my solution was to clear the sync cache after every sync.
I'm running a syncing script with cron which looks something like this:

/usr/local/bin/bwdc login CLIENT_ID CLIENT_SECRET
/usr/local/bin/bwdc sync
/usr/local/bin/bwdc clear-cache
/usr/local/bin/bwdc logout

Don't know if you need to login and logout every time but that worked for me.

**Reply by Sir-IT-Papi** [reply]
Hey! just wanted to let you know that I tested this out and yeah, this works! I have to clear cache and sign out and sign back in and it seems to work! I've tested it out manually via the app as I don't have a script like you set up! so now is this bug with the app or with vault?

**Reply by jobritz** [reply]
I'm not sure, I just found this solution during my tests, but I have to use the CLI Version because my host server does not have a GUI installed and the CLI Version is easy to automate on linux with cron or on windows with a scheduled task. I only used the Desktop App for the configuration.
